src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Transaction transaction) {
+        Fruit fruit = transaction.getFruit();
+        Integer currentQuantity = Storage.get(fruit);
+        Storage.put(fruit, Optional.of(currentQuantity - transaction.getQuantity())
```

Comment:
if currentQuantity is null, `currentQuantity - transaction.getQuantity()` will produce NPE

check if it is null or if the quantity is insufficient
if any answer is yes - throw exception